project/Dependencies.scala

```diff
@@ -1,6 +1,11 @@
 import sbt._
+import sbt.Keys._
 
 object Dependencies {
+  val mill = Def.setting {
+    val version = if (scalaBinaryVersion.value == "2.12") "0.6.3" else "0.7.4"
```

Comment:
Just some food for thought: I think at compile time you should bind to the lowest API denominator, `0.6.0` resp. `0.7.0`. And only use newer versions at runtime. (Can't tell you if and how that is possible with sbt.)

Reply:
Good point. We should do that in a separate PR I think.